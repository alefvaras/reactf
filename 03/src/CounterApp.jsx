import { useState } from 'react'
import PropTypes from 'prop-types';
export const CounterApp = ({ value }) => {
    const [counter, setCounter] = useState(value);

 
    const handleAdd = () => {
        // setCounter(counter+1) 
        //c es el valor actual del counter, cuando desconocemos cual es
        setCounter((c) => c + 1)

    };
    const handleRest = () => {
        setCounter(counter - 1)

    };

    const reset = () => {
        setCounter(value);
        //setCounter((c) => c = value)


    };
    

    return (

        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
              {/* <button onClick={(event)=>handleAdd(event)}>+1</button> */}
            <button onClick={handleAdd}>+1</button>
            <button onClick={handleRest}>-1</button>
            <button aria-label="btn-reset" onClick={reset}>Reset</button>

        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

