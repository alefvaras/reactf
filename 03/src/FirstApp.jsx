
const newMessage = {
    message: 'hola',
    name: 'antonio'
};

import { PropTypes } from "prop-types";
const getMessage = () => 'hola yoda';
// export const Name = ({ title, subtitulo="holi subtitulo" }) => {
    export const Name = ({ title, subtitulo }) => {
//PropTypes define el tipo a las props
// if(!title){
//     throw new Error('tile no no envia')
// }
    return (
        <>
            <h1 data-testid="test-title">{title}</h1>
            {/* <div>{JSON.stringify(newMessage)}</div> */}
            <p>{subtitulo }</p>
            <p>{getMessage()}</p>
        </>

    )

}
Name.propTypes={
title:PropTypes.string.isRequired
}

Name.propTypes={
    //entran antes  del  propTypes
}
Name.defaultProps={
    subtitulo:'holi'
    }