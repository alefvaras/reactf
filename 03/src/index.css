html, body{
    background-color: black;
    color: aliceblue;
    font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
    font-size: 1.3rem;
}

button{
    padding: 15px;
    font-size: 1.3rem;
    margin-right: 5px;
}