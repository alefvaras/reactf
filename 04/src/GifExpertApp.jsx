import React, { useState } from 'react'
import { AddCategory,GifGrid } from './components';


const apiKey = '9MUKydmrQOEkfBWWSPUOYkrka2ORDcjy';

export const GifExpertApp = () => {

  const [categories, setcategories] = useState(['dragon ball'])
  const onAddCategory = (newCategory) => {
    //  setcategories([...categories,'ale']);
 
    if(categories.includes(newCategory)) return;
    setcategories(cat => {
      return [...cat, newCategory]
    });
  }

  return (
    <>
      <h1>GifExpertApp</h1>
      {/* <AddCategory setcategories={setcategories} /> */}
      <AddCategory onNewCategory={value=>onAddCategory(value)}/>
      {/* <button onClick={onAddCategory}>Agregar</button> */}

        {categories.map(category => 

           (
            <GifGrid key={category} category={category}/>
        
          )
        )}
 
    </>

  )
}
