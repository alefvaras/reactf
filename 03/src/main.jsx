import React from 'react';
import ReactDOM from 'react-dom/client';
import { HelloWorld } from './HelloWorld';
 import  { Name } from './FirstApp';
 import './index.css'
import { CounterApp } from './CounterApp';

ReactDOM.createRoot(document.getElementById("root")).render(
    <React.StrictMode>
        {/* <Name title={123}  subtitulo={1+9}/> */}
        {/* <Name  subtitulo={1+9}/>
        <Name title subtitulo={1+9}/> */}
        {/* <Name  subtitulo={1+9}/> */}
        {/* <Name title />
        <HelloWorld/> */}
       <CounterApp value={100}/>
   
   {/* <Name title="titulo"/> */}
    </React.StrictMode>
) 
