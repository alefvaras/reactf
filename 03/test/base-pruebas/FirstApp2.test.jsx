import { render, screen } from "@testing-library/react";
import { Name } from "../../src/FirstApp";

describe('App2', () => {
    const title = "titulo";
    const subtitulo="subtitulo";
    test('debe hacer match con el snapshot', () => {

        const { container } = render(<Name title={title} />)
        expect(container).toMatchSnapshot();
    });
    test('debe de mostrar el titulo', () => {
        render(<Name title={title} />)
        //ultima version actualizada despues de los cambios del DOM
        //  screen
        expect(screen.getByText(title)).toBeTruthy();
    });
    test('retornar el titulo h1', () => {
        render(<Name title={title} />)
        expect(screen.getByRole("heading", { level:1 }).innerHTML).toContain(title);
    });
    test('debe de mostrar subtitulo por props', () => { 
        render(<Name title={title} subtitulo="subtitulo" />)
        expect(screen.getAllByText(subtitulo).length).toBe(1);
     })
})