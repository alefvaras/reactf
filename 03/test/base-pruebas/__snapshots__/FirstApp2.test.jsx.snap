// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App2 debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    titulo
  </h1>
  <p>
    holi
  </p>
  <p>
    hola yoda
  </p>
</div>
`;
